pkg/payload/precondition/clusterversion/upgradeable.go

```diff
@@ -51,7 +51,7 @@ func ClusterVersionOverridesCondition(cv *configv1.ClusterVersion) *configv1.Clu
 // Run runs the Upgradeable precondition.
 // If the feature gate `key` is not found, or the api for clusterversion doesn't exist, this check is inert and always returns nil error.
 // Otherwise, if Upgradeable condition is set to false in the object, it returns an PreconditionError when possible.
-func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.ReleaseContext, clusterVersion *configv1.ClusterVersion) error {
+func (pf *Upgradeable) Run(ctx context.Context, releaseContext precondition.ReleaseContext) error {
```

Comment:
Test this carefully. The reason I originally added `clusterVersion`, pulled in `Start`, was that it was always set to the true, "current" (Completed) version, whereas `cv`, pulled during precondition evaluation, was showing up as the "desired" target version. Not to say things have not changed or that my initial testing that lead down that path was not flawed somehow but worth a word of caution.

Reply:
moving this conversation from the function signature [to the one consumer][1], so it's more clear how the value is being used (or was being used, before my proposed change in this PR).

[1]: https://github.com/openshift/cluster-version-operator/pull/708#discussion_r766334993